catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityFilterPlugin.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.metacard.validation;
+
+import static ddf.catalog.metacard.validation.MetacardValidityMarkerPlugin.VALIDATION_ERRORS;
+import static ddf.catalog.metacard.validation.MetacardValidityMarkerPlugin.VALIDATION_WARNINGS;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PostQueryPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+
+public class MetacardValidityFilterPlugin implements PostQueryPlugin {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(MetacardValidityFilterPlugin.class);
+
+    private static Map<String, List<String>> attributeMap = new HashMap<>();
+
+    public static final String INVALID_STATE = "invalid_state";
+
+    @Override
+    public QueryResponse process(QueryResponse input)
+            throws PluginExecutionException, StopProcessingException {
+        List<Result> results = input.getResults();
+
+        // process security
+        for (Result result : results) {
+            Metacard metacard = result.getMetacard();
+            Attribute securityAttribute = metacard.getAttribute(Metacard.SECURITY);
+            HashMap<String, List<String>> securityMap = null;
+            Serializable tempSecurityMap = null;
+            if (securityAttribute != null) {
+                tempSecurityMap = securityAttribute.getValue();
+                try {
+                    securityMap = (HashMap<String, List<String>>) tempSecurityMap;
+
+                } catch (Exception e) {
```

Comment:
Also you should set securitymap to the default value ifyou threw an exception so its not null